include/fs/nvs.h

```diff
@@ -102,8 +99,9 @@ int nvs_clear(struct nvs_fs *fs);
  * @param data Pointer to the data to be written
  * @param len Number of bytes to be written
  *
- * @return Number of bytes written. On success, it will be equal to the number
- * of bytes requested to be written. On error returns -ERRNO code.
+ * @return Number of bytes written. On success, it will be equal to the number of bytes requested
+ * to be written. When a rewrite of the same data already stored is attempted, nothing is written
+ * to flash, thus 0 is returned. On error returns -ERRNO code.
```

Comment:
We should probably replace the `-ERRNO` with something like `one of negative errno.h defined error codes`, or something like this, to make it more clear.